src/QMCWaveFunctions/Fermion/SlaterDetBuilder.h

```diff
@@ -100,6 +101,27 @@ class SlaterDetBuilder : public WaveFunctionComponentBuilder
                      bool& optimizeCI,
                      int nels_up,
                      int nels_dn);
+
+  template<typename VT, std::enable_if_t<std::is_floating_point<VT>::value, int> = 0>
+  void readCoeffs(hdf_archive& hin, std::vector<VT>& ci_coeff, size_t n_dets)
```

Comment:
How about
```
void readCoeffs(hdf_archive& hin, std::vector<RealType>& ci_coeff, size_t n_dets)
void readCoeffs(hdf_archive& hin, std::vector<std::complex<RealType>>& ci_coeff, size_t n_dets)
```
So we don't need to write the function in header and we can skip the slow SFINAE.

Reply:
I see your point as its currently written.
  
I've finished my thought on this with this next push.  I like this better because its expresses an  logically inconsistent for a SlaterDetBuilder with ValueType = std:complex<...> to have a readCoeffs that takes std::vector<RealType> and attempting to call it is a logic error that can be caught at compile time. In this case I think the SFINAE impact should be negligible since this is used from one compilation unit.

Reply:
I see your point as its currently written.

I've finished my thought on this with this next push. I like this better because its expresses an logically inconsistent for a SlaterDetBuilder with ValueType = std:complex<...> to have a readCoeffs that takes std::vector and attempting to call it is a logic error that can be caught at compile time. In this case I think the SFINAE impact should be negligible since this is used from one compilation unit.